### ISSUE TITLE
`*.sancov` files not generated when fuzzing woff2

### ISSUE THREAD

**Issue by prashast** [google/fuzzer-test-suite#146]
I wanted to get the sancov coverage files generated using the default ASAN hooks for the guard functions and was invoking it by first building with the appropriate flags inside Docker using the base image `libfuzzertutorial/base`:
```
# ~/FTS/woff2-2016-05-06/build.sh asan
# ASAN_OPTIONS=coverage=1 ./woff2-2016-05-06-libfuzzer
```

The fuzzer by itself is working fine and I can see it discovering new paths. However, no sancov files are generated which should be if the guard hooks were being hit. Is there anything that I might be doing wrong?

**Comment by kcc** [comment]
This functionality has been deleted from libFuzzer a while ago, sorry. 
If you build w/o libFuzzer it should still work though. (but we don't try very hard to maintain it)

**Comment by prashast** [comment]
Aah, I see. Yeah without libfuzzer, I can confirm it works. So I assume, the way to get the coverage data while using LibFuzzer would be to put in my own hooks as described in https://github.com/google/fuzzer-test-suite/blob/master/tutorial/example-hooks.cc and compile it in with the fuzz target?

I was even trying to use the `clang-coverage` as recommended in the readme by using these set of flags to compile Woff with libfuzzer: `-O2 -fno-omit-frame-pointer -g -fprofile-instr-generate -fcoverage-mapping` using the custom-build script. However, I ended up with some undefined references inside the `libfuzzer.a` archive. 
```
libFuzzingEngine-libfuzzer.a(FuzzerDriver.o): In function `thread<void (&)()>':
/usr/lib/gcc/x86_64-linux-gnu/5.4.0/../../../../include/c++/5.4.0/thread:137: undefined reference to `pthread_create'
libFuzzingEngine-libfuzzer.a(FuzzerDriver.o): In function `thread<void (&)(const fuzzer::Command &, std::atomic<unsigned int> *, unsigned int, std::atomic<bool> *), std::reference_wrapper<fuzzer::Command>, std::atomic<unsigned int> *, unsigned int &, std::atomic<bool> *>':
/usr/lib/gcc/x86_64-linux-gnu/5.4.0/../../../../include/c++/5.4.0/thread:137: undefined reference to `pthread_create'
libFuzzingEngine-libfuzzer.a(FuzzerDriver.o): In function `thread<void (&)(fuzzer::Fuzzer *, unsigned long), fuzzer::Fuzzer *&, unsigned long &>':
/usr/lib/gcc/x86_64-linux-gnu/5.4.0/../../../../include/c++/5.4.0/thread:137: undefined reference to `pthread_create'
libFuzzingEngine-libfuzzer.a(FuzzerShmemPosix.o): In function `fuzzer::SharedMemoryRegion::Create(char const*)':
/root/test/../Fuzzer/FuzzerShmemPosix.cpp:53: undefined reference to `sem_unlink'
/root/test/../Fuzzer/FuzzerShmemPosix.cpp:54: undefined reference to `sem_open'
/root/test/../Fuzzer/FuzzerShmemPosix.cpp:53: undefined reference to `sem_unlink'
/root/test/../Fuzzer/FuzzerShmemPosix.cpp:54: undefined reference to `sem_open'
libFuzzingEngine-libfuzzer.a(FuzzerShmemPosix.o): In function `fuzzer::SharedMemoryRegion::Open(char const*)':
/root/test/../Fuzzer/FuzzerShmemPosix.cpp:72: undefined reference to `sem_open'
/root/test/../Fuzzer/FuzzerShmemPosix.cpp:72: undefined reference to `sem_open'
libFuzzingEngine-libfuzzer.a(FuzzerShmemPosix.o): In function `fuzzer::SharedMemoryRegion::Post(int)':
/root/test/../Fuzzer/FuzzerShmemPosix.cpp:86: undefined reference to `sem_post'
libFuzzingEngine-libfuzzer.a(FuzzerShmemPosix.o): In function `fuzzer::SharedMemoryRegion::Wait(int)':
/root/test/../Fuzzer/FuzzerShmemPosix.cpp:91: undefined reference to `sem_wait'
/root/test/../Fuzzer/FuzzerShmemPosix.cpp:91: undefined reference to `sem_wait'
```

**Comment by kcc** [comment]
you need -lpthread to resolve these

**Comment by prashast** [comment]
Thanks for that input. I added the flag and it was able to compile successfully for generating `clang-coverage` reports. However, while running the compiled binary, even though the libfuzzer works fine and as expected with the `woff2` seed inputs it finds new paths during the fuzz run, on termination of the fuzzing run it generates a single empty `default.profdata` file. Are there some other steps that I may be missing? I also made sure the docker image was started with the `--cap-add=SYS_PTRACE` functionality. 

Sorry for the persistent questioning and thanks for taking time out for answering these queries.
